[Interaction] Convert ToAU Missions

claywar opened the pull request:
<!-- remove space and place 'x' mark between square [] brackets or click the checkbox after saving to affirm: -->
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] that I agree to LandSandBoat's [Limited Contributor License Agreement](https://github.com/LandSandBoat/server/blob/base/.github/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I have **read the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)**
- [x] that I've _**tested my code and things my code changed**_ since the last commit in the PR, and will test after any later commits

This PR converts TOAU missions up to 45 to Interaction Framework.  The remaining few missions will require more recent an thorough caps in order to implement properly, so have been removed from scope, and will be handled once sufficient data is collected.

claywar committed:
Interaction - Mission: Immortal Sentries

claywar committed:
Interaction - Mission: President Salaheem

claywar committed:
Interaction - Mission: Knight of Gold

claywar committed:
Interaction - Mission: Confessions of Royalty

claywar committed:
Interaction - Mission: Easterly Winds

claywar committed:
Interaction - Mission: Westerly Winds

claywar committed:
Interaction - Mission: A Mercenary Life

claywar committed:
Remove duplicated code for PFC and SP Wildcat promotions

claywar committed:
Interaction - Mission: Undersea Scouting

claywar committed:
Interaction - Mission: Astral Waves

claywar committed:
Interaction - Mission: Imperial Schemes

claywar committed:
Interaction - Mission: Royal Puppeteer

claywar committed:
Interaction - Mission: Lost Kingdom

claywar committed:
Interaction - Mission: The Dolphin Crest

claywar committed:
Interaction - Mission: The Black Coffin

claywar committed:
Interaction - Mission: Ghosts of the Past

Also adjusts Whitegate Shared.lua to return an object, and removes dORoyalPalaceArmorCheck from CI ignore list

KnowOne134 commented on scripts/missions/toau/03_President_Salaheem.lua:
```diff
@@ -0,0 +1,61 @@
+                    local missionStatus = player:getMissionStatus(mission.areaId)
+
+                    if missionStatus == 0 then
+                        return mission:progressEvent(73, 0, 0, 0, 0, 0, 0, 0, 0, 0)
```
To make these cleaner instead of a bunch of 0's all the time could be
(73, {[8] = 0}) or (73, text_table = 0) depending on what version of code you have

claywar committed:
Merge remote-tracking branch 'origin/base' into interaction/toau

claywar committed:
Interaction - Mission: Guests of the Empire


Additional header comments

claywar committed:
Interaction - Mission: Passing Glory

Confirmed this CS is not an onTrigger action, but region on the first ramp approaching Naja

claywar committed:
Interaction - Mission: Sweets for the Soul

claywar committed:
Interaction - Mission: Teahouse Tumult


Fix logic error

claywar committed:
Interaction - Mission: Finders Keepers


Remove title, as this is granted in the next mission

claywar committed:
Interaction - Mission: Shield of Diplomacy

claywar committed:
Interaction - Mission: Social Graces

claywar committed:
Interaction - Mission: Foiled Ambition

claywar committed:
Interaction - Mission: Playing the Part

claywar committed:
Interaction - Mission: Seal of the Serpent

claywar committed:
Interaction - Mission: Misplaced Nobility

claywar committed:
Interaction - Mission: Bastion of Knowledge


QC: Add battlefieldWin check to Shield of Diplomacy

claywar committed:
Interaction - Mission: Puppet in Peril


Fix Puppet in Peril title comment

claywar committed:
Interaction - Mission: Prevalence of Pirates


Comment formatting

claywar committed:
Interaction - Mission: Shades of Vengeance

Also updates instance to add registry/entry requirements, and uses global onInstanceCreatedCallback

Add Escape Temp Item, and message for KI Fading

claywar committed:
Interaction - Mission: In the Blood

Also adds Naja Salaheem location comment to M25 (Playing the Part)

claywar committed:
Interaction - Mission: Sentinels' Honor

claywar committed:
Interaction - Mission: Testing the Waters

Xaver-DaRed committed:
[Fix] Azouph Staging Point + incorrect unlocks (#1070)

* Change HXI client version for November Update

* Fix Staging Points

* Fix trailing whitespace

* Re-Delete original Runic portal script

* Simplify check in runic portals

* Remove Runic Portals from ToAU Immortal_Sentries script

Co-authored-by: claywar <81713309+claywar@users.noreply.github.com>

claywar committed:
Interaction - Mission: Legacy of the Lost

claywar committed:
Interaction - Mission: Gaze of the Saboteur

Also removes unnecessary require in Legacy of the Lost

Fix comments

claywar committed:
Interaction - Mission: Path of Blood

claywar committed:
Interaction - Mission: Stirrings of War

claywar committed:
Interaction - Mission: Allied Rumblings

claywar committed:
Interaction - Mission: Unraveling Reason

claywar committed:
Interaction - Mission: Light of Judgement

claywar committed:
Interaction - Mission: Path of Darkness

Includes minor cleanup including KI naming convention for "Fades," and temporarily reverting text_table references.

claywar committed:
Interaction - Mission: Fangs of the Lion


Fix missing return in Path of Darkness

claywar committed:
Interaction - Mission: Nashmeira's Plea

claywar committed:
Interaction - Mission: Ragnarok

zach2good committed:
[core] Correct name lookup for instance_filename in LoadEventScript

claywar committed:
Merge remote-tracking branch 'origin/interaction/toau' into interaction/toau2

claywar committed:
Merge remote-tracking branch 'origin/base' into interaction/toau2

Xaver-DaRed committed:
ToAU - Mission 2 and 3 fixes.

Xaver-DaRed committed:
Minor corrections to missions 2 and 3

Xaver-DaRed committed:
ToAU Mission 16

claywar committed:
Merge pull request #1105 from Xaver-DaRed/interaction/toau-fixes

[Interaction] ToAU Missions 2 and 3 fixes.

claywar committed:
Merge remote-tracking branch 'origin/base' into interaction/toau3

claywar committed:
Merge pull request #1109 from Xaver-DaRed/Fix/Mission16

[Interaction] ToAU Mission 16 and 17 fixes

Xaver-DaRed committed:
Fix issues with ToAU missions 4 to 13

claywar committed:
Merge pull request #1164 from Xaver-DaRed/interaction/toau-fixes-2

[Interaction] ToAU mission 4 to 13 fixes

claywar committed:
Merge remote-tracking branch 'origin/base' into toau

claywar committed:
Merge remote-tracking branch 'origin/base' into interaction/toau

claywar committed:
Clean IDs descriptions to conform with tooling


Mamool Ja string

claywar committed:
Update 9-argument event strings to use text_table

claywar committed:
Merge remote-tracking branch 'origin/base' into interaction/toau

claywar committed:
Merge remote-tracking branch 'origin/base' into interaction/toau

claywar committed:
QC: ToAU Missions


Tweak Salaheem's Sentinels regions to prevent overlap


Add missing DB entries for Puppet in Peril NPCs


Use default instance event finish and update functions


Remove errant reward from TOAU33

claywar committed:
Merge remote-tracking branch 'origin/base' into interaction/toau

Xaver-DaRed commented on scripts/missions/toau/40_Unraveling_Reason.lua:
```diff
@@ -0,0 +1,108 @@
+                    if option == 1 then
+                        -- Title can be obtained immediately following this event.  Since these are stringed
+                        -- together, set it at the end for clarity in mission rewards.
+                        player:setPos(0, 0, 0, 0, 51)
```
Wouldnt this pos get overwritten to default pos defined in Zone.lua? if so, and becouse of certain PRs like this #1222 shouldnt we set the actual pos we want just on case?

claywar replied:
Its there on purpose to trigger a download, the onZoneIn return forces the correct series of events (in the mission script)

Xaver-DaRed replied:
yeah, i get that, I'm not arguing against it. but i meant the first 3 values that determine the x y z.

the 5th value is the zone id, witch when set forces a zone change no matter if the target zone is the same as current zone.

So what I meant is, shouldn't we be setting the correct x,y,z just so the "onZoneIn invalid pos fix" wont kick in if the target pos in said zone is ever changed?

claywar replied:
In that case, we really don't care where we go in the zone, because there's several back-to-back events that are triggered in sequence.  We're forcing a zone change, and then handling it on the back end (they're all on zone ins, so a disconnect would resume where it left off)

Xaver-DaRed commented on src/map/lua/luautils.cpp:
```diff
@@ -1,4 +1,4 @@
-﻿/*
+/*
```
theres a lot of files with changes in this line. Is this adding that red dot thing?

zach2good replied:
Byte order mark stuff, I don't remember how we deal with this

claywar committed:
QC: TOAU Missions 2


Add additional comments to TOAU40
